### PR TITLE
rest/rpc: Make mempoolinfo atomic

### PR DESCRIPTION
**Pull request by promag**
Make `/rest/mempool/info.json` endpoint and `getmempoolinfo` RPC atomic.

### CODE REVIEW

**Comment by geekastute** [comment]
sounds good but need to review more thoroughly

**Comment by MarcoFalke** [comment]
So what is the point of locking in the public mempool calls in the first place? If there is no reason, I'd rather remove those locks and pass the responsibility (and flexibility) to lock to the caller.

**Comment by promag** [comment]
@MarcoFalke I was going that way but the change started to get big. Since it's a recursive mutex (for now) this is also correct.

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1487,6 +1487,7 @@ static UniValue getchaintips(const JSONRPCRequest& request)
 
 UniValue mempoolInfoToJSON()
```
Comment (MarcoFalke) [inline src/rpc/blockchain.cpp]: Needs rebase

Comment (promag) [reply src/rpc/blockchain.cpp]: Done.

File: src/rpc/blockchain.cpp [hunk 2]
```diff
@@ -1483,6 +1483,7 @@ static UniValue getchaintips(const JSONRPCRequest& request)
 
 UniValue MempoolInfoToJSON(const CTxMemPool& pool)
 {
+    LOCK(::mempool.cs);
```
Comment (MarcoFalke) [inline src/rpc/blockchain.cpp]:
```suggestion
    LOCK(pool.cs);
```

Comment (promag) [reply src/rpc/blockchain.cpp]: Duh ofc, thanks1

**Pull request merged**